pkg/autoupdate/autoupdate.go

```diff
@@ -88,23 +88,24 @@ func New(
 }
 
 // Run runs the autoupdate controller.
-func (ctrl *Controller) Run(ctx context.Context, workers int) {
+func (ctrl *Controller) Run(ctx context.Context, workers int) error {
 	defer utilruntime.HandleCrash()
 	defer ctrl.queue.ShutDown()
 
 	klog.Info("Starting AutoUpdateController")
 	defer klog.Info("Shutting down AutoUpdateController")
 
 	if !cache.WaitForCacheSync(ctx.Done(), ctrl.cacheSynced...) {
-		klog.Info("Caches never synchronized")
-		return
+		return fmt.Errorf("caches never synchronized: %w", ctx.Err())
```

Comment:
I don't think it ever gives up.

Reply:
> I don't think it ever gives up.

```console
$ git --no-pager grep -hA3 '// WaitForCacheSync' vendor/k8s.io/client-go/tools/cache/shared_informer.go
// WaitForCacheSync waits for caches to populate.  It returns true if it was successful, false
// if the controller should shutdown
// callers should prefer WaitForNamedCacheSync()
func WaitForCacheSync(stopCh <-chan struct{}, cacheSyncs ...InformerSynced) bool {
```

We should respect their advertised API (and that's what we're doing).  If they never give up, that's up to them, and an upstream bug if it's a problem for some reason.